src/Xamarin.Android.Build.Tasks/Tasks/GenerateRtxt.cs

```diff
@@ -15,21 +10,32 @@ public class GenerateRtxt : AndroidTask
 	{
 		public override string TaskPrefix => "GR";
 
+		public bool DesignTimeBuild { get; set; }
+
 		[Required]
 		public string RTxtFile { get; set; }
 
 		[Required]
 		public string ResourceDirectory { get; set; }
 		public string[] AdditionalResourceDirectories { get; set; }
 
-		[Required]
 		public string JavaPlatformJarPath { get; set; }
 
 		public string ResourceFlagFile { get; set; }
 		public string CaseMapFile { get; set; }
 
 		public override bool RunTask ()
 		{
+			if (string.IsNullOrEmpty (JavaPlatformJarPath)) {
```

Comment:
So this is needed so we can find the `public.xml` document. This helps us fixup any `android:xxx` style resources with the correct ids values. 

For a design time build if might be better to just default to `0`. But we will need to make sure this does not leak into a non design time build otherwise things will break. 

see https://github.com/dotnet/android/blob/862923b2511c87e894af63f1edd5fec2d3977020/src/Xamarin.Android.Build.Tasks/Utilities/FileResourceParser.cs#L35

In fact we already handle the case where we don't have a publicxml https://github.com/dotnet/android/blob/main/src/Xamarin.Android.Build.Tasks/Utilities/FileResourceParser.cs#L291. 
So it might be enough to take the required off. 

But we do need to make sure if we then get a patform.jar we regenerate the designer.